### PR TITLE
Use SHOULD .. can rather than MAY

### LINKED ISSUE #4559: Barry Leiba's Transport Comment 2
Barry Leiba said:

> — Section 9.6.2 —
>    It SHOULD drop packets
>    for this connection received on the old IP address, but MAY continue
>    to process delayed packets.
> 
> Do as you think best with cases such as these, but for my part, I dislike the
> “SHOULD... but MAY” formulation, as I find it contradictory when it’s read
> strictly.  In general, I prefer to simply avoid the BCP 14 key word for the
> second part (“SHOULD do x, but may make other choices”).  In both cases here,
> I’d probably just leave off the second part, as it doesn’t seem to add
> anything.  At the least, I encourage making it “may” (or “can”).  But again, as
> you think best.﻿

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #4559.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -800,7 +800,7 @@ any outstanding data is declared lost, the endpoint SHOULD send a RESET_STREAM
 frame instead of retransmitting the data.
 
 An endpoint SHOULD copy the error code from the STOP_SENDING frame to the
-RESET_STREAM frame it sends, but MAY use any application error code.  An
+RESET_STREAM frame it sends, but can use any application error code.  An
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:803]: Drop the "but can..." -- it's not necessary

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -2705,7 +2705,7 @@ Once the server has completed its path validation and has received a non-probing
 packet with a new largest packet number on its preferred address, the server
 begins sending non-probing packets to the client exclusively from its preferred
 IP address.  It SHOULD drop packets for this connection received on the old IP
-address, but MAY continue to process delayed packets.
+address, but can continue to process delayed packets.
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:2708]: This is badly written -- the SHOULD and the MAY apply to different packets. How about:
"The server SHOULD drop newer packets for this connection that are received on the old IP address. The server MAY continue to process older delayed packets that are received on the old IP address."

**Review by janaiyengar — CHANGES_REQUESTED** [review]

File: draft-ietf-quic-transport.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-quic-transport.md:2708]: Please make a separate issue and pull request for this.  I agree that this is a better formulation.

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (janaiyengar) [reply draft-ietf-quic-transport.md:803]: My point here is that removing the MAY is ok, but the semantics are different. "can" is a statement of fact, and it's not necessary to say that an an endpoint _can_ send any error code in addition to what it SHOULD do. Though I don't feel strongly.

If the can is to be retained, I'd suggest flipping the primary and secondary clauses -- "can use any error code, but SHOULD copy"

**Review by janaiyengar — APPROVED** [review]
Ok, let's merge this one, and I'll make a separate PR for the second comment.

**Pull request merged**